src/cfi.rs

```diff
@@ -454,12 +456,734 @@ impl<'input, Endian> FrameDescriptionEntry<'input, Endian>
+            }
+
+            // Row push and pop instructions.
+            RememberState => {
+                self.ctx.push_row();
+            }
+            RestoreState => {
+                assert!(self.ctx.stack.len() > 0);
+                if self.ctx.stack.len() == 1 {
+                    return Err(Error::PopWithEmptyStack);
+                }
+                self.ctx.pop_row();
+            }
+
+            // No operation.
+            Nop => {}
+        };
+
+        Ok(false)
+    }
+}
+
+/// A row in the virtual unwind table that describes how to find the values of
+/// the registers in the *previous* frame for a range of PC addresses.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct UnwindTableRow<'input, Endian>
+    where Endian: Endianity
+{
+    start_address: u64,
+    end_address: u64,
+    cfa: CfaRule<'input, Endian>,
+    registers: Vec<RegisterRule<'input, Endian>>,
+}
+
+impl<'input, Endian> Default for UnwindTableRow<'input, Endian>
+    where Endian: Endianity
+{
+    fn default() -> Self {
+        UnwindTableRow {
+            start_address: 0,
+            end_address: 0,
+            cfa: Default::default(),
+            registers: Vec::new(),
+        }
+    }
+}
+
+impl<'input, Endian> UnwindTableRow<'input, Endian>
+    where Endian: Endianity
+{
+    /// Get the starting PC address that this row applies to.
+    pub fn start_address(&self) -> u64 {
+        self.start_address
+    }
+
+    /// Get the end PC address where this row's register rules become
+    /// unapplicable.
+    ///
+    /// In other words, this row describes how to recover the last frame's
+    /// registers for all PCs where `row.start_address() <= PC <
+    /// row.end_address()`. This row does NOT describe how to recover registers
+    /// when `PC == row.end_address()`.
+    pub fn end_address(&self) -> u64 {
+        self.end_address
+    }
+
+    /// Return `true` if the given `address` is within this row's address range,
+    /// `false` otherwise.
+    pub fn contains(&self, address: u64) -> bool {
+        self.start_address <= address && address < self.end_address
+    }
+
+    /// Get the canonical frame address (CFA) recovery rule for this row.
+    pub fn cfa(&self) -> &CfaRule<'input, Endian> {
+        &self.cfa
+    }
+
+    /// Get the register recovery rules for this row.
+    ///
+    /// The rule at index `i` of the slice is the rule for register `i`. The
+    /// slice is **not** guaranteed to have length equal to the number of
+    /// registers on the target architecture. It is only as long as the largest
+    /// numbered register for which a recovery rule is defined in this row. All
+    /// others are implied to be `Undefined`.
+    ///
+    /// The register number mapping is architecture dependent. For example, in
+    /// the x86-64 ABI the register number mapping is defined in Figure 3.36:
+    ///
+    /// > Figure 3.36: DWARF Register Number Mapping
```

Comment:
In practice, I think library consumbers need these register number mappings to make heads and tails of the resulting unwind table rows.

Should we make a `register_number_mapping` module exposing constant structs like `register_number_mapping::x86_64`?